### PR TITLE
Swap undef checks for strict-variable-checking-compatible checks

### PR DESCRIPTION
**Pull request by ross-w**
Hello there

I believe this fixes issue #82 by asking defined rather than trying to compare with a potentially undef variable. Comments welcome.

### CODE REVIEW

**Comment by sammcj** [comment]
Have tested @ross-w's branch against 4 CentOS 7 machines of various state and it fixes all the puppet 4 variable issues!

++ on merging this

**Comment by alexjfisher** [comment]
This at least partially duplicates https://github.com/voxpupuli/puppet-gluster/pull/83

File: manifests/peer.pp [hunk 1]
```diff
@@ -45,7 +45,7 @@
   $pool = 'default'
 ) {
 
-  $binary = $::gluster_binary
+  $binary = defined('$gluster_binary') ? { true => $::gluster_binary, false => false }
```
Comment (alexjfisher) [inline manifests/peer.pp]: @ross-w Thanks for your PR and the time you've taken to submit it.

Unfortunately using `defined()` for checking facts like this is broken on puppet versions before 3.7.5.
I've added a spec test and done a bit of playing around in https://github.com/voxpupuli/puppet-gluster/pull/85 to show this.

Hopefully I'll find time to get a final PR (one without my extra 'test' commits) submitted (and merged) tomorrow.

**Comment by alexjfisher** [comment]
https://github.com/voxpupuli/puppet-gluster/issues/87 was merged instead.

**Pull request closed without merging**